models/linkages/R/model2netcdf.LINKAGES.R

```diff
@@ -20,144 +20,116 @@
 ##' @param end_date End time of the simulation
 ##' @export
 ##'
-##' @author Ann Raiho # I changed this a bunch to do the MIP runs... Might need to change it back. Didn't commit additions for MIP runs.
+##' @author Ann Raiho, Betsy Cowdery
 model2netcdf.LINKAGES <- function(outdir, sitelat, sitelon, start_date=NULL, end_date=NULL,force=FALSE){ #, PFTs) {
 #  logger.severe("NOT IMPLEMENTED")
-  
-  # I'm bringing this back until we come up with a better solution. 
-  PFTs = c("acer","betula","carya","castanea.dentata","fagus.grandifolia","picea","pinus","tsuga.canadensis","quercus")
-  # PFTs <- as.character(c(38, 72, 58, 8, 2, 1, 6, 7, 11)) these aren't the same.
 
-  ### Read in model output in LINKAGES format
-  output <- as.matrix(read.csv(file.path(outdir, "OUT.csv"), header=FALSE))
-  
-  output[is.na(output)]<-0
-  
-  output.dims <- dim(output)    
-  block.size = round(output.dims[1]/4)-1 #change if you change kprint
-  LINKAGES.output = matrix(numeric(0),block.size,output.dims[2]*2 - 1)
-  LINKAGES.output[,1:output.dims[2]] = output[1:block.size,]
-  LINKAGES.output[,(1:(output.dims[2]-1)) + output.dims[2] ] = output[(1:block.size) + block.size,-1]
- 
-  LINKAGES.pft = matrix(numeric(0),block.size,output.dims[2]*2 - 3)
-  LINKAGES.pft[,1:(output.dims[2] - 1) ] = output[1+(1:block.size) + 2*block.size,c(-11)]
-  LINKAGES.pft[,(1:(output.dims[2] -2)) + (output.dims[2] - 1) ] = output[1+(1:block.size) + 3*block.size,c(-1,-11)]
+  require(PEcAn.utils)
+  require(ncdf4)
   
-  #LINKAGES.output = LINKAGES.output[,-c(31,41)]
-  
-  colnames(LINKAGES.output) <- c("year","numStems","agBiomass","leafLitter","leafLitterN","agNPP","availN","humusCN","soilResp","soilOM","ET",
-                                 "numStems.SD","agBiomass.SD","leafLitter.SD","leafLitterN.SD","agNPP.SD","availN.SD","humusCN.SD","soilResp.SD","soilOM.SD","ET.SD")
-  LINKAGES.output <- as.data.frame(LINKAGES.output)
-  LINKAGES.output$year <- seq(0,1160,2)
+  ##QUESTIONS:How should I name the PFTs? What should I do about more than one plot?
```

Comment:
PFT names in the output should ideally match the names in PEcAn. PEcAn outputs right now are formulated at the grid cell level so if you have multiple plots the overall output is typically their mean. That said, we're definitely increasing the number of models capable of producing patch/plot scale output (ED, LINKAGES, CLM-ED, LPJ-GUESS) so we should discuss a way of saving and naming this info that's generic across models